src/test/java/lambda/part1/exercise/Exercise3.java

```diff
@@ -0,0 +1,71 @@
+package lambda.part1.exercise;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+import lambda.data.Person;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.arrayContaining;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+class Exercise3 {
+
+    @Test
+    void sortPersonsByAgeUsingArraysSortExpressionLambda() {
+        Person[] persons = getPersons();
+
+        // TODO use Arrays.sort + expression-lambda
+        Arrays.sort(persons, (p1, p2) -> p1.getAge() - p2.getAge());
+
+        assertThat(persons, is(arrayContaining(
+                new Person("Иван", "Мельников", 20),
+                new Person("Николай", "Зимов", 30),
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45)
+        )));
+    }
+
+    @Test
+    @SuppressWarnings("all")
+    void sortPersonsByLastNameThenFirstNameUsingArraysSortExpressionLambda() {
+        Person[] persons = getPersons();
+
+        // TODO use Arrays.sort + statement-lambda
+        Arrays.sort(persons, (p1, p2) -> {
+            int lastNameCompareResult = p1.getLastName().compareTo(p2.getLastName());
+            return lastNameCompareResult == 0 ? p1.getFirstName().compareTo(p2.getFirstName()) : lastNameCompareResult;
+        });
+
+        assertThat(persons, is(arrayContaining(
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45),
+                new Person("Николай", "Зимов", 30),
+                new Person("Иван", "Мельников", 20)
+        )));
+    }
+
+    @Test
+    @SuppressWarnings("all")
+    void findFirstWithAge30UsingGuavaPredicateLambda() {
+        List<Person> persons = Arrays.asList(getPersons());
+
+        // TODO use FluentIterable
+        Person person = FluentIterable.from(persons).filter((p) -> p.getAge() == 30).first().get();
```

Comment:
This way is slower then firstMatch, because you require iterator to go through all of the elements

Reply:
Thank you for advice, search improved.